Classes/TestableFramework.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * (c) Contributors of the Neos Project - www.neos.io
+ * Please see the LICENSE file which was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\BuildEssentials;
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Core\Bootstrap;
+use Neos\Flow\Core\RequestHandlerInterface;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Utility\Files;
+
+/**
+ * Low level operations for use in testing, all to be used in static calls.
+ *
+ * @Flow\Proxy(false)
+ */
+abstract readonly class TestableFramework
+{
+    /**
+     * Calculates a FLOW_ROOTPATH
+     * @return string
+     */
+    public static function calculateRootPath(): string
+    {
+        $flowRootPathFromEnvironment = trim(getenv('FLOW_ROOTPATH') ?: '', '"\' ');
+        return $flowRootPathFromEnvironment ?: dirname(__DIR__, 3) . '/';
+    }
+
+    public static function getApplicationContext(): string
+    {
+        $flowContextFromEnvironment = trim(getenv('FLOW_CONTEXT') ?: '', '"\' ');
+        return $flowContextFromEnvironment ?: 'Testing';
+    }
+
+    /**
+     * Build a Flow Bootstrap with an instance of the given class implementing RequestHandlerInterface
+     *
+     * @param string $context
+     * @param class-string<RequestHandlerInterface> $requestHandlerClassname
+     * @return Bootstrap
+     */
+    public static function buildBootstrapWithPreselectedRequestHandler(string $context, string $requestHandlerClassname): Bootstrap
+    {
+        $flowBootstrap = new Bootstrap($context);
+        $requestHandlerInstance = new $requestHandlerClassname($flowBootstrap);
+        assert($requestHandlerInstance instanceof RequestHandlerInterface);
+
+        $flowBootstrap->registerRequestHandler($requestHandlerInstance);
+        $flowBootstrap->setPreselectedRequestHandlerClassName($requestHandlerClassname);
+        return $flowBootstrap;
+    }
+
+    /**
+     * Creates an empty table structure according to the active ORM schemas
+     *
+     * @param Bootstrap $bootstrap
+     * @return void
+     * @throws \Doctrine\ORM\Tools\ToolsException
+     * @throws \Neos\Flow\Exception
+     */
+    public static function enablePersistence(Bootstrap $bootstrap): void
```

Comment:
i actually dont know if its a good idea to put this highly internal Neos.Flow persistence booting logic in here ... for example its not really reusable for our behat stuff:

https://github.com/neos/behat/blob/f09b2b3ecd7907cf8f1e63a46f84300ac4df2860/Classes/FlowEntitiesTrait.php#L35

and i think it should rather reside as close as possible to Neos.Flow so it can be 'fixed' or simplified at some point.